pkg/model/interface.go

```diff
@@ -138,25 +150,52 @@ func (v *VLLMParam) DeepCopy() VLLMParam {
 
 // builds the container command:
 // eg. torchrun <TORCH_PARAMS> <OPTIONAL_RDZV_PARAMS> baseCommand <MODEL_PARAMS>
-func (p *PresetParam) GetInferenceCommand(runtime RuntimeName, skuNumGPUs string, configVolume *corev1.VolumeMount) []string {
+func (p *PresetParam) GetInferenceCommand(runtime RuntimeName, gpuConfig *sku.GPUConfig, configVolume *corev1.VolumeMount) []string {
 	switch runtime {
 	case RuntimeNameHuggingfaceTransformers:
 		torchCommand := utils.BuildCmdStr(p.Transformers.BaseCommand, p.Transformers.TorchRunParams, p.Transformers.TorchRunRdzvParams)
 		modelCommand := utils.BuildCmdStr(p.Transformers.InferenceMainFile, p.Transformers.ModelRunParams)
 		return utils.ShellCmd(torchCommand + " " + modelCommand)
 	case RuntimeNameVLLM:
-		if p.VLLM.ModelName != "" {
-			p.VLLM.ModelRunParams["served-model-name"] = p.VLLM.ModelName
-		}
-		if !p.DisableTensorParallelism {
-			p.VLLM.ModelRunParams["tensor-parallel-size"] = skuNumGPUs
-		}
-		if configVolume != nil {
-			p.VLLM.ModelRunParams["kaito-config-file"] = path.Join(configVolume.MountPath, ConfigfileNameVLLM)
-		}
-		modelCommand := utils.BuildCmdStr(p.VLLM.BaseCommand, p.VLLM.ModelRunParams)
-		return utils.ShellCmd(modelCommand)
+		return p.getInferenceCommandForVLLM(gpuConfig, configVolume)
 	default:
 		return nil
 	}
 }
+
+// builds the container command for vllm runtime
+// notes: all these pre-rendered parameters can be replaced by user-specified config
```

Comment:
I added a note here. If "gpu-memory-utilization" is provided in both cli flag and configmap, configmap one will be the source of true. @Fei-Guo